doc-generator/src/main/scala/codacy/pylint/Main.scala

```diff
@@ -84,39 +91,117 @@ object Main {
+  final case class Parameter(name: String, description: String, default: Value)
+
+  val parameters = Map[String, Seq[Parameter]](
+    "R0914" -> Seq(
+      Parameter(
+        "max-locals",
+        "Maximum number of locals for function / method body.",
+        15
+      )
+    ),
+    "C0301" -> Seq(
+      Parameter(
+        "max-line-length",
+        "Maximum number of characters on a single line.",
+        120
+      )
+    ),
+    "C0102" -> Seq(
+      Parameter(
+        "bad-names",
+        "Bad variable names which should always be refused, separated by a comma.",
+        "foo,bar,baz,toto,tutu,tata"
+      )
+    ),
+    "C0103" ->
+      Seq(
+        Parameter(
+          "argument-rgx",
+          "Regular expression matching correct argument names. Overrides argument- naming-style.",
+          "[a-z_][a-z0-9_]{2,30}$"
+        ),
+        Parameter(
+          "attr-rgx",
+          "Regular expression matching correct attribute names.",
+          "[a-z_][a-z0-9_]{2,30}$"
+        ),
+        Parameter(
+          "class-rgx",
+          "Regular expression matching correct class names.",
+          "[A-Z_][a-zA-Z0-9]+$"
+        ),
+        Parameter(
+          "const-rgx",
+          "Regular expression matching correct constant names.",
+          "(([A-Z_][A-Z0-9_]*)|(__.*__))$"
+        ),
+        Parameter(
+          "function-rgx",
+          "Regular expression matching correct function names.",
+          "[a-z_][a-z0-9_]{2,30}$"
+        ),
+        Parameter(
+          "method-rgx",
+          "Regular expression matching correct method names.",
+          "[a-z_][a-z0-9_]{2,30}$"
+        ),
+        Parameter(
+          "module-rgx",
+          "Regular expression matching correct module names.",
+          "(([a-z_][a-z0-9_]*)|([A-Z][a-zA-Z0-9]+))$"
+        ),
+        Parameter(
+          "variable-rgx",
+          "Regular expression matching correct variable names.",
+          "[a-z_][a-z0-9_]{2,30}$"
+        ),
+        Parameter(
+          "inlinevar-rgx",
+          "Regular expression matching correct inline iteration names.",
+          "[A-Za-z_][A-Za-z0-9_]*$"
+        ),
+        Parameter(
+          "class-attribute-rgx",
+          "Regular expression matching correct class attribute names.",
+          "([A-Za-z_][A-Za-z0-9_]{2,30}|(__.*__))$"
         )
+      )
+  )
+
+  def addPatternsParameters(obj: Obj, ruleName: String): Unit =
+    for {
+      params <- parameters.get(ruleName)
+    } {
+      obj("parameters") = params.map(
+        param => Obj("name" -> param.name, "default" -> param.default)
+      )
+    }
+
+  def addDescriptionParameters(obj: Obj, ruleName: String): Unit =
+    for {
+      params <- parameters.get(ruleName)
+    } {
+      obj("parameters") = params.map(
+        param => Obj("name" -> param.name, "description" -> param.description)
+      )
     }
-  }
```

Comment:
I would separate the `for` into another function, so we don't have duplicate code.

Something like this:

```
def addPatternsParameters(obj: Obj, ruleName: String): Unit =
    addParameters(
      obj,
      ruleName,
      param => Obj("name" -> param.name, "default" -> param.default)
    )

  def addDescriptionParameters(obj: Obj, ruleName: String): Unit =
    addParameters(
      obj,
      ruleName,
      param => Obj("name" -> param.name, "description" -> param.description)
    )

  def addParameters(
      obj: Obj,
      ruleName: String,
      transformation: Parameter => Obj
  ): Unit =
    for {
      params <- parameters.get(ruleName)
    } {
      obj("parameters") = params.map(transformation)
    }
```